src/classes/DiscordBot.ts

```diff
@@ -0,0 +1,77 @@
+/* eslint-disable @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-call, @typescript-eslint/no-unsafe-assignment */
+
+import { Client, Intents, Message } from 'discord.js';
+import log from '../lib/logger';
+import Options from './Options';
+import Bot from './Bot';
+import SteamID from 'steamid';
+
+export default class DiscordBot {
+    readonly client: Client;
+
+    constructor(private options: Options, private bot: Bot) {
+        this.client = new Client({
+            intents: [Intents.FLAGS.GUILDS, Intents.FLAGS.GUILD_MESSAGES, Intents.FLAGS.DIRECT_MESSAGES]
+        });
+        void this.client.login(this.options.discordApiToken);
```

Comment:
Maybe put the login not in the constructor (in the `start` instead).
If `login` is a Promise, then it should be handled accordingly.